Data/Aeson/Types/ToJSON.hs

```diff
@@ -2716,20 +2718,24 @@ packChunks lbs =
 
 --------------------------------------------------------------------------------
 
+-- | Wrap a list of pairs as an object.
 class Monoid pairs => FromPairs enc pairs | enc -> pairs where
   fromPairs :: pairs -> enc
 
-instance FromPairs Encoding Series where
+instance (a ~ Value) => FromPairs (Encoding' a) Series where
   fromPairs = E.pairs
 
 instance FromPairs Value (DList Pair) where
   fromPairs = object . toList
 
-class Monoid kv => GKeyValue v kv where
-    gPair :: String -> v -> kv
+-- | Like 'KeyValue' but the value is already converted to JSON
+-- ('Value' or 'Encoding'), and the result actually represents lists of pairs
+-- so it can be readily concatenated.
+class Monoid kv => KeyValuePair v kv where
+    pair :: String -> v -> kv
 
-instance ToJSON v => GKeyValue v (DList Pair) where
-    gPair k v = DList.singleton (pack k .= v)
+instance (v ~ Value) => KeyValuePair v (DList Pair) where
```

Comment:
Is there a reason this isn't `KeyValuePair Value (DList Pair)` ?

Reply:
It improves type inference. Otherwise the TH deriving results in ambiguous type variables and we would need more type annotations.